Lib9c/Battle/CPHelper.cs

```diff
@@ -12,14 +12,39 @@ namespace Nekoyume.Battle
     public static class CPHelper
     {
         public static int TotalCP(
-            IEnumerable<Equipment> equipments,
-            IEnumerable<Costume> costumes,
-            IEnumerable<RuneOptionSheet.Row.RuneOptionInfo> runeOptions,
+            IReadOnlyCollection<Equipment> equipments,
+            IReadOnlyCollection<Costume> costumes,
+            IReadOnlyCollection<RuneOptionSheet.Row.RuneOptionInfo> runeOptions,
             int level,
             CharacterSheet.Row row,
-            CostumeStatSheet costumeStatSheet)
+            CostumeStatSheet costumeStatSheet, List<StatModifier> collectionStatModifiers)
         {
-            var levelStatsCp = GetStatsCP(row.ToStats(level), level);
+            decimal levelStatsCp;
+            var collectionCp = 0m;
+            // CharacterStats.BaseStats CP equals row.ToStats
+            if (collectionStatModifiers.Any())
+            {
+                // Prepare CharacterStats for calculate collection Stats
+                var characterStats = new CharacterStats(row, level);
+                characterStats.SetEquipments(equipments, new EquipmentItemSetEffectSheet());
+                characterStats.SetCostumeStat(costumes, costumeStatSheet);
+                foreach (var runeOption in runeOptions)
+                {
+                    characterStats.AddRuneStat(runeOption);
+                }
+
+                characterStats.SetCollections(collectionStatModifiers);
+                levelStatsCp = GetStatsCP(characterStats.BaseStats, level);
+                foreach (var (statType, value) in characterStats.CollectionStats.GetStats())
+                {
+                    collectionCp += GetStatCP(statType, value);
+                }
+            }
+            else
+            {
+                levelStatsCp = GetStatsCP(row.ToStats(level), level);
+            }
```

Comment:
```suggestion
            decimal levelStatsCp = GetStatsCP(row.ToStats(level), level);
            var collectionCp = 0m;
            // CharacterStats.BaseStats CP equals row.ToStats
            if (collectionStatModifiers.Any())
            {
                // Prepare CharacterStats for calculate collection Stats
                var characterStats = new CharacterStats(row, level);
                characterStats.SetEquipments(equipments, new EquipmentItemSetEffectSheet());
                characterStats.SetCostumeStat(costumes, costumeStatSheet);
                foreach (var runeOption in runeOptions)
                {
                    characterStats.AddRuneStat(runeOption);
                }

                characterStats.SetCollections(collectionStatModifiers);
                foreach (var (statType, value) in characterStats.CollectionStats.GetStats())
                {
                    collectionCp += GetStatCP(statType, value);
                }
            }
```
levelStetsCP를 굳이 따로 계산할필요가 없어보여서 제안드립니다.